database/factories/AlbumFactory.php

```diff
@@ -1,79 +1,96 @@
-
-/*
-|--------------------------------------------------------------------------
-| Model Factories
-|--------------------------------------------------------------------------
-|
-| This directory should contain each of the model factory definitions for
-| your application. Factories provide a convenient way to generate new
-| model instances for testing / seeding your application's database.
-|
-*/
-
-$factory->define($album->class(), function (Faker $faker) use ($artist, $song) {
-
-    return [
-        'artist_id' => function() use ($artist) {
-            return factory($artist->class())->create()->id;
-        },
-        'title' => $faker->company,
-        'alt_title' => $faker->company,
-        'year' => $faker->year('now'),
-        'description' => $faker->sentence(10),
-        'has_explicit_lyrics' => $faker->boolean(25),
-        'is_active' => $faker->boolean(80),
-    ];
-});
-
-$factory->state($album->class(), 'withSongs', function ($faker) use ($song) {
-    return [
-        'songs' => factory($song->class(), rand(1, 20))->make(['is_active' => 1]),
-    ];
-});
-
-$factory->afterCreating($album->class(), function ($album, $faker) use ($song) {
-
-    // Create and associate some Songs.
-
-    for ($i = 1; $i < rand(2, 21); $i++) {
-        $s = factory($song->class())->create([
-            'album_id' => $album->id,
-            'track_number' => $i,
-            'is_active' => 1,
-        ]);
-
-        $s->album()->associate($album)->save();
-    }
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Str;
+use IndieHD\Velkart\Contracts\Repositories\Eloquent\ProductRepositoryContract;
+
+class AlbumFactory extends Factory
+{
+    /**
+     * The name of the factory's corresponding model.
+     *
+     * @var string
+     */
+    protected $model = Album::class;
+
+    public function configure()
+    {
+        return $this->afterCreating(function (Album $album) {
+
+            // Create and associate some Songs.
+
+            $songs = rand(2, 21);
+
+            for ($i = 1; $i < $songs; $i++) {
+                static::factoryForModel(resolve(SongRepositoryInterface::class)->class())->create([
+                    'album_id' => $album->id,
+                    'track_number' => $i,
+                    'is_active' => 1,
+                ]);
+            }
 
-    // Create and associate a Digital Asset.
+            // Create and associate a Digital Asset.
 
-    $album->asset()->save(factory(DigitalAsset::class)->make([
-        'product_id' => factory(Product::class)->create([
-            'name' => $album->title,
-            'slug' => Str::slug($album->title),
-            'description' => $album->description,
-            'price' => $album->full_album_price ?? ($album->songs->count() * 100),
-        ])->id,
-        'asset_id' => $album->id,
-        'asset_type' => App\Album::class,
-    ]));
+            $album = $album->fresh(['songs']);
```

Comment:
Good catch. I recall seeing a note about this in Velkart, too:

https://github.com/indiehd/velkart/issues/20

I'm not necessarily suggesting you fix it there, right this moment, but this is the same issue/reason for the change, correct?

Reply:
Correct same reason. Hadn't seen the issue, but it is true.